### PR TITLE
Created option drop down menu for selection list

### PR DESCRIPTION
**Pull request by jasonto510**
Deleted a lot of the comments that were there previously. I created a drop down menu with all of the possible options, still need to refactor to find the best possible way to get a drop down menu for each selection instead of one drop down menu with each option.

### CODE REVIEW

File: client/src/Components/SelectionList.jsx [hunk 1]
```diff
@@ -7,85 +7,72 @@ class SelectionList extends React.Component {
...
-  //Afterwards can get the options with the same method
+  getOptionsValue(){
+    var options = [];
+    return getItemSelection(this.props.randomItemNumber)
+      .then((data) => {
+        console.log(data, 'data from options value')
+        if (data[0]['options']){
+          for (var key in data[0]['options']){
+            var individualOptions = [];
+            for (let i = 0; i < data[0]['options'][key].length; i++){
```
Comment (JonBeyond) [inline client/src/Components/SelectionList.jsx]: Strongly recommend when you have large variable access names such as `data[0]['options'][key]` that you assign it to a easily readable variable name the line above where you need to use it.  For example, you could say `let option = data[0]['options'][key];` and then refer to it just as option in the `for` loop and in the `push` operator

it's definitely an art form for when to do this versus not, my general rule of thumb is if it's hard to read, I assign it a temporary variable so it's easy to determine whats going on

EDIT: Another alternative, you could have assigned a variable to `let varname = data[0]['options']` instead, and then just referred to that as you go down the chain, since you are rewriting that long term many times.  Later on you could say `varname[key]` for example.  Up to you on how you want to refactor this

File: client/src/Components/SelectionList.jsx [hunk 2]
```diff
@@ -7,85 +7,72 @@ class SelectionList extends React.Component {
...
-  //     return;
-    
-  //   })
-  //   console.log(selection[0], 'after return statement')  
-  //   return selection;
-  // }
 
-  getItemOptions(){
-    console.log(this.state.selectionPossibilties)
+  getSelectValue(){
```
Comment (JonBeyond) [inline client/src/Components/SelectionList.jsx]: I am going to place an 'overall review' comment regarding the functions `getSelectValue` and `getOptionValue`

**Review by JonBeyond — CHANGES_REQUESTED** [review]
Two main points:
`point 1: code reuse`
I strongly recommend that you refactor your functions `getSelectValue` and `getOptionsValue`.  At the very least, you should only send a single `GET` request to your server->database.  As it stands right now, you are sending two equal queries to your server for the same data.  You have a few options 1) you could store the entire contents of the data on the client (i.e. as a state variable or better yet, a stateless variable)
2) you could combine these two functions together and process the data that is returned at the same time

`point 2: not all options/selections are showing`
This is a related comment, but it may give you more ideas on how to solve the above (or disregard it completely and change how you're handling your data filtering).  This is specifically regarding your initial question in the pull request (**not all options/selection combos are showing**). 

You are currently storing what were multiple key:value pairs in your database into two separate arrays.
If your database is already storing all of your data in the form of objects, where each key is the option name, and the value is an array of the selections... then... can you just pull this out of your database and store it in a single state variable? I believe it roughly looks like this:

```
options: {
'options1': [selections...],
'options2': [selections...]... etc
}
```

To summarize, it appears as though you are deconstructing your database, which is already in the format you want, and then trying to piece it back together using arrays, which is going to be complicated (it's doable, just very messy and would require some other variables to keep track of indexes for each option, yuck).

Then you could pass the single prop down to your module and have it use a `for...in` loop to process each key and a `forEach` or `for..of` and render each selection.

I'm going to deny this merge.  While functional, it's not production ready.  Specifically, I would deny this on the sole grounds that you're querying the database twice for the same information.
